torch/csrc/jit/codegen/cuda/scheduler/registry.cpp

```diff
@@ -9,12 +9,15 @@ namespace fuser {
 namespace cuda {
 
 bool SchedulerEntry::sameAs(const SchedulerEntry* other) {
-  if (has_param_ != other->has_param_) {
+  if (has_reduction_param_ != other->has_reduction_param_) {
     return false;
   }
-  if (has_param_) {
+  if (has_reduction_param_) {
     return rparams_ == other->rparams_;
+  } else {
+    return pparams_ == other->pparams_;
   }
```

Comment:
This makes sure the caching is safe. We may want to update the hash as well. It's `size_t SchedulerEntryHash::operator()(const SchedulerEntry& se) const {` at around line 362.

Reply:
Just fixed this, will push. Thanks.